File: apps/api/app/Http/Controllers/AdminPengajuanController.php
<?php

namespace App\Http\Controllers;

use App\Models\Pengajuan;
use Illuminate\Http\Request;

class AdminPengajuanController extends Controller
{
    public function index(Request $request)
    {
        $query = Pengajuan::query();

        if ($request->has('search') && $request->search) {
            $search = $request->search;
            $query->where(function ($q) use ($search) {
                $q->where('nama', 'LIKE', "%{$search}%")
                  ->orWhere('nim', 'LIKE', "%{$search}%")
                  ->orWhere('jenis_berkas', 'LIKE', "%{$search}%");
            });
        }

        if ($request->has('status') && $request->status) {
            $query->where('status', $request->status);
        }

        return response()->json(
            $query->orderBy('created_at', 'desc')->paginate($request->get('per_page', 10))
        );
    }

    public function show($id)
    {
        $pengajuan = Pengajuan::findOrFail($id);
        return response()->json($pengajuan);
    }

    public function updateProgress(Request $request, $id)
    {
        $validated = $request->validate([
            'status' => 'required|string|in:Diproses,Ditandatangani,Selesai,Ditolak',
            'catatan_admin' => 'nullable|string',
        ]);

        $pengajuan = Pengajuan::findOrFail($id);
        $pengajuan->status = $validated['status'];
        if (isset($validated['catatan_admin'])) {
            $pengajuan->catatan_admin = $validated['catatan_admin'];
        }
        $pengajuan->save();

        return response()->json([
            'message' => 'Progress berhasil diupdate',
            'data' => $pengajuan
        ]);
    }

    public function destroy($id)
    {
        $pengajuan = Pengajuan::findOrFail($id);
        $pengajuan->delete();

        return response()->json(['message' => 'Pengajuan berhasil dihapus']);
    }

    public function stats()
    {
        $total = Pengajuan::count();
        $diproses = Pengajuan::where('status', 'Diproses')->count();
        $ditandatangani = Pengajuan::where('status', 'Ditandatangani')->count();
        $selesai = Pengajuan::where('status', 'Selesai')->count();
        $ditolak = Pengajuan::where('status', 'Ditolak')->count();

        // Calculate Hari Ini and Perhatian
        $hari_ini = Pengajuan::whereDate('created_at', now()->toDateString())->count();
        $perhatian = Pengajuan::where('status', 'Diproses')->count();

        // Recent 7 days stats
        $weeklyStats = [];
        for ($i = 6; $i >= 0; $i--) {
            $date = now()->subDays($i)->format('Y-m-d');
            $weeklyStats[] = [
                'date' => now()->subDays($i)->format('D'), // Mon, Tue, etc.
                'count' => Pengajuan::whereDate('created_at', $date)->count()
            ];
        }

        // Generate dynamic activities based on recent pengajuan
        $recentPengajuans = Pengajuan::orderBy('updated_at', 'desc')->take(4)->get();
        $activities = [];
        foreach ($recentPengajuans as $p) {
            $timeAgo = $p->updated_at->diffForHumans();
            
            if ($p->status === 'Selesai') {
                $user = 'Admin';
                $action = "menyelesaikan pengajuan {$p->jenis_berkas} milik {$p->nama}";
                $icon = 'check_circle';
                $iconColor = 'text-[#10b981]';
                $bgColor = 'bg-[#10b981]/10';
            } elseif ($p->status === 'Ditandatangani') {
                $user = 'Dekan Fakultas Sains';
                $action = "menandatangani berkas {$p->jenis_berkas} milik {$p->nama}";
                $icon = 'edit';
                $iconColor = 'text-primary';
                $bgColor = 'bg-primary/10';
            } elseif ($p->status === 'Ditolak') {
                $user = 'Admin';
                $action = "menolak pengajuan {$p->jenis_berkas} milik {$p->nama}";
                $icon = 'error';
                $iconColor = 'text-error';
                $bgColor = 'bg-error/10';
            } else { // Diproses
                $user = $p->nama;
                $action = "mengirim pengajuan baru: {$p->jenis_berkas}";
                $icon = 'description';
                $iconColor = 'text-blue-500';
                $bgColor = 'bg-blue-500/10';
            }

            $activities[] = [
                'user' => $user,
                'action' => $action,
                'time' => $timeAgo,
                'icon' => $icon,
                'iconColor' => $iconColor,
                'bgColor' => $bgColor
            ];
        }

        return response()->json([
            'total' => $total,
            'diproses' => $diproses,
            'ditandatangani' => $ditandatangani,
            'selesai' => $selesai,
            'ditolak' => $ditolak,
            'hari_ini' => $hari_ini,
            'perhatian' => $perhatian,
            'weekly' => $weeklyStats,
            'activities' => $activities
        ]);
    }
}

File: apps/api-java/src/main/java/com/fspengajuan/api/controller/AdminPengajuanController.java
package com.fspengajuan.api.controller;

import com.fspengajuan.api.model.Pengajuan;
import com.fspengajuan.api.repository.PengajuanRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.time.Duration;

@RestController
@RequestMapping("/api/admin/pengajuan")
public class AdminPengajuanController {

    private final PengajuanRepository pengajuanRepository;

    public AdminPengajuanController(PengajuanRepository pengajuanRepository) {
        this.pengajuanRepository = pengajuanRepository;
    }

    @GetMapping
    public ResponseEntity<?> index(
            @RequestParam(required = false) String search,
            @RequestParam(required = false) String status,
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(defaultValue = "10") int per_page) {

        Pageable pageable = PageRequest.of(page - 1, per_page, Sort.by(Sort.Direction.DESC, "createdAt"));
        Page<Pengajuan> result;

        if (search != null && !search.isEmpty() && status != null && !status.isEmpty()) {
            result = pengajuanRepository.searchAllAndStatus(search, status, pageable);
        } else if (search != null && !search.isEmpty()) {
            result = pengajuanRepository.searchAll(search, pageable);
        } else if (status != null && !status.isEmpty()) {
            result = pengajuanRepository.findByStatus(status, pageable);
        } else {
            result = pengajuanRepository.findAll(pageable);
        }

        // Format to match Laravel pagination
        Map<String, Object> response = new HashMap<>();
        response.put("data", result.getContent());
        response.put("current_page", result.getNumber() + 1);
        response.put("total", result.getTotalElements());
        response.put("last_page", result.getTotalPages());
        response.put("per_page", result.getSize());

        return ResponseEntity.ok(response);
    }

    @GetMapping("/{id}")
    public ResponseEntity<?> show(@PathVariable Long id) {
        return pengajuanRepository.findById(id)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    @PutMapping("/{id}/progress")
    public ResponseEntity<?> updateProgress(@PathVariable Long id, @RequestBody Map<String, String> request) {
        return pengajuanRepository.findById(id).map(pengajuan -> {
            if (request.containsKey("status")) {
                pengajuan.setStatus(request.get("status"));
            }
            if (request.containsKey("catatan_admin")) {
                pengajuan.setCatatanAdmin(request.get("catatan_admin"));
            }
            pengajuanRepository.save(pengajuan);
            
            Map<String, Object> response = new HashMap<>();
            response.put("message", "Progress berhasil diupdate");
            response.put("data", pengajuan);
            return ResponseEntity.ok(response);
        }).orElse(ResponseEntity.notFound().build());
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<?> destroy(@PathVariable Long id) {
        if (pengajuanRepository.existsById(id)) {
            pengajuanRepository.deleteById(id);
            Map<String, String> response = new HashMap<>();
            response.put("message", "Pengajuan berhasil dihapus");
            return ResponseEntity.ok(response);
        }
        return ResponseEntity.notFound().build();
    }

    @GetMapping("/stats")
    public ResponseEntity<?> stats() {
        long total = pengajuanRepository.count();
        long diproses = pengajuanRepository.countByStatus("Diproses");
        long ditandatangani = pengajuanRepository.countByStatus("Ditandatangani");
        long selesai = pengajuanRepository.countByStatus("Selesai");
        long ditolak = pengajuanRepository.countByStatus("Ditolak");

        LocalDateTime startOfDay = LocalDate.now().atStartOfDay();
        LocalDateTime endOfDay = LocalDate.now().plusDays(1).atStartOfDay();
        long hariIni = pengajuanRepository.countByCreatedAtBetween(startOfDay, endOfDay);
        long perhatian = diproses;

        List<Map<String, Object>> weeklyStats = new ArrayList<>();
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("E");
        for (int i = 6; i >= 0; i--) {
            LocalDate date = LocalDate.now().minusDays(i);
            LocalDateTime start = date.atStartOfDay();
            LocalDateTime end = date.plusDays(1).atStartOfDay();
            
            Map<String, Object> stat = new HashMap<>();
            stat.put("date", date.format(formatter));
            stat.put("count", pengajuanRepository.countByCreatedAtBetween(start, end));
            weeklyStats.add(stat);
        }

        List<Pengajuan> recentPengajuans = pengajuanRepository.findTop4ByOrderByUpdatedAtDesc();
        List<Map<String, Object>> activities = new ArrayList<>();
        
        for (Pengajuan p : recentPengajuans) {
            Duration duration = Duration.between(p.getUpdatedAt(), LocalDateTime.now());
            String timeAgo = duration.toHours() > 0 ? duration.toHours() + " hours ago" : duration.toMinutes() + " minutes ago";
            
            String user, action, icon, iconColor, bgColor;
            if ("Selesai".equals(p.getStatus())) {
                user = "Admin";
                action = "menyelesaikan pengajuan " + p.getJenisBerkas() + " milik " + p.getNama();
                icon = "check_circle";
                iconColor = "text-[#10b981]";
                bgColor = "bg-[#10b981]/10";
            } else if ("Ditandatangani".equals(p.getStatus())) {
                user = "Dekan Fakultas Sains";
                action = "menandatangani berkas " + p.getJenisBerkas() + " milik " + p.getNama();
                icon = "edit";
                iconColor = "text-primary";
                bgColor = "bg-primary/10";
            } else if ("Ditolak".equals(p.getStatus())) {
                user = "Admin";
                action = "menolak pengajuan " + p.getJenisBerkas() + " milik " + p.getNama();
                icon = "error";
                iconColor = "text-error";
                bgColor = "bg-error/10";
            } else {
                user = p.getNama();
                action = "mengirim pengajuan baru: " + p.getJenisBerkas();
                icon = "description";
                iconColor = "text-blue-500";
                bgColor = "bg-blue-500/10";
            }
            
            Map<String, Object> activity = new HashMap<>();
            activity.put("user", user);
            activity.put("action", action);
            activity.put("time", timeAgo);
            activity.put("icon", icon);
            activity.put("iconColor", iconColor);
            activity.put("bgColor", bgColor);
            activities.add(activity);
        }

        Map<String, Object> response = new HashMap<>();
        response.put("total", total);
        response.put("diproses", diproses);
        response.put("ditandatangani", ditandatangani);
        response.put("selesai", selesai);
        response.put("ditolak", ditolak);
        response.put("hari_ini", hariIni);
        response.put("perhatian", perhatian);
        response.put("weekly", weeklyStats);
        response.put("activities", activities);

        return ResponseEntity.ok(response);
    }
}
